OpenSim/Common/ComponentSocket.h

```diff
@@ -721,6 +721,19 @@ class Input : public AbstractInput {
         SimTK_INDEXCHECK_ALWAYS(index, getNumConnectees(),
                                 "Input<T>::setAlias()");
 
+        const auto& connecteeName = getConnecteeName(index);
+        std::string ignore1{}, ignore2{}, ignore3{};
+        std::string currAlias{};
+        parseConnecteeName(connecteeName,
+                           ignore1, ignore2, ignore3, currAlias);
+        if(currAlias.empty()) {
+            setConnecteeName(getConnecteeName(index) + "(" + alias + ")",
```

Comment:
It seems that is would be possible to get `/path_to_output/out_name|channel_name()(alias)` since there isn't an explicit check that the brackets are present.  I think it would be nice if there was a `composeConnecteeName` analog to the parse method with the same arguments and you simply passed in the new alias (last argument). It would be a lot more readable and easy to adapt if/when we move to `ChannelPath`